addons/hr_work_entry/models/hr_work_entry.py

```diff
@@ -66,6 +67,52 @@ class HrWorkEntry(models.Model):
     def init(self):
         tools.create_index(self._cr, "hr_work_entry_date_start_date_stop_index", self._table, ["date_start", "date_stop"])
 
+    def _get_actual_durations(self):
+        return sum(work_entry.duration for work_entry in self)
+
+    def _get_calendar(self):
+        return self.employee_id.resource_calendar_id or self.company_id.resource_calendar_id
+
+    def _get_work_entries_data(self, work_entries_dict):
+        data = self.env['hr.work.entry']
+        for key, work_entries in work_entries_dict.items():
+            actual_duration = work_entries._get_actual_durations()
```

Comment:
the name of this variable is also confusing. What actual duration? What it really is the full day duration. Maybe `whole_day_duration` or `full_day_duration`. Or `duration_on_the_day`. Any of it will do